lib/passport-configurator.js

```diff
@@ -294,31 +294,32 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
                     function(err, accessToken) {
                       if (err) {
                         return err.code === 'LOGIN_FAILED'
-                            ? done(null, false, {message: 'Login failed'})
+                            ? done(null, false, {message: errorMsg})
                             : done(err);
                       }
-                      if (accessToken) {
+                      if (accessToken && user.emailVerified) {
                         userProfile.accessToken = accessToken;
                         done(null, userProfile, {accessToken: accessToken});
                       } else {
-                        done(null, false, {message: 'Failed to create token.'});
+                        done(null, false, {message: errorMsg});
                       }
                     });
                 }
-              }
+              } else {
```

Comment:
that `else` is the main fix for the bug. if we remove it, we will get the error message of ```Error: Can't set headers after they are sent.```